test/k8sT/Services.go

```diff
@@ -1547,6 +1548,143 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
+			// }
+
+			ciliumPodK8s1, err := kubectl.GetCiliumPodOnNode(helpers.K8s1)
+			Expect(err).Should(BeNil(), "Cannot get cilium pod on k8s1")
+			monitorRes1, monitorCancel1 = kubectl.MonitorStart(ciliumPodK8s1)
+			ciliumPodK8s2, err := kubectl.GetCiliumPodOnNode(helpers.K8s2)
+			Expect(err).Should(BeNil(), "Cannot get cilium pod on k8s2")
+			monitorRes2, monitorCancel2 = kubectl.MonitorStart(ciliumPodK8s2)
+
+			// Wait for all client and server pods to be in ready state.
+			err = kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=affinity-server", helpers.HelperTimeout)
+			Expect(err).Should(BeNil())
+			// Wait for the service backends to be plumbed in BPF maps
+			podIPs, err := kubectl.GetPodsIPs(helpers.DefaultNamespace, serverPodLabel)
+			ExpectWithOffset(1, err).Should(BeNil(), "Cannot retrieve pod IPs for %s", serverPodLabel)
+			ExpectWithOffset(1, len(podIPs)).To(Equal(2), "Expected pod IPs mismatch")
+			for _, ip := range podIPs {
+				err = kubectl.WaitForServiceBackend(helpers.K8s1, ip)
+				ExpectWithOffset(1, err).Should(BeNil(), "Failed waiting for %s backend entry on k8s1", ip)
+				err = kubectl.WaitForServiceBackend(helpers.K8s2, ip)
+				ExpectWithOffset(1, err).Should(BeNil(), "Failed waiting for %s backend entry on k8s2", ip)
+			}
+			err = kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=affinity-client", helpers.HelperTimeout)
+			Expect(err).Should(BeNil())
+			pods, err := kubectl.GetPodNames(helpers.DefaultNamespace, clientPodLabel)
+			ExpectWithOffset(1, err).Should(BeNil(), "cannot retrieve pod names by filter %q", clientPodLabel)
+			clientPod = pods[0]
+			ctx, cancel := context.WithCancel(context.Background())
+			res = kubectl.LogsStream(helpers.DefaultNamespace, clientPod, ctx)
+			// Check if the client pod is able to connect to the server once it's up and running
+			Expect(res.WaitUntilMatch("client successfully connected")).To(BeNil(),
+				"%s is not in the output after timeout", res.GetStdOut())
+
+			defer func() {
+				cancel()
+				res.WaitUntilFinish()
+				helpers.WriteToReportFile(res.CombineOutput().Bytes(), "affinity-client-pod-logs.txt")
+			}()
+		})
+
+		AfterFailed(func() {
+			kubectl.CiliumReport("cilium service list", "cilium bpf lb list")
+			kubectl.LogsWithLabel(helpers.DefaultNamespace, serverPodLabel)
+			kubectl.LogsWithPrevious(helpers.DefaultNamespace, clientPod)
+			monitorCancel1()
+			monitorCancel2()
+			helpers.WriteToReportFile(monitorRes1.CombineOutput().Bytes(), "affinity-cilium-monitor-k8s1.log")
+			helpers.WriteToReportFile(monitorRes2.CombineOutput().Bytes(), "affinity-cilium-monitor-k8s2.log")
+		})
+
+		AfterAll(func() {
+			_ = kubectl.Delete(affinitySvcYAML)
+			_ = kubectl.Delete(affinityServerYAML)
+			_ = kubectl.Delete(affinityClientYAML)
+			// if helpers.DualStackSupported() {
+			// 	_ = kubectl.Delete(affinitySvc6YAML)
+			// }
+		})
+
+		It("Checks clusterIP connectivity is uninterrupted on service endpoints' updates", func() {
```

Comment:
Our code to generate Jenkins artifacts often struggles with special characters. We could spend more time fixing it, but TBH it's easier to just rename the test :-)
```suggestion
		It("Checks clusterIP connectivity is uninterrupted on service endpoint updates", func() {
```